src/transforms/groupby.js

```diff
@@ -118,7 +128,7 @@ function transformOne(trace, state) {
         }
 
         newTrace.name = groupName;
-        newTrace.marker.color = opts.groupColors[groupName];
+        newTrace = Lib.extendDeep(newTrace, style[groupName] || {});
```

Comment:
... but maybe add a comment mentioning there's no need to coerce `style[groupName]` here as another round of `supplyDefaults` is done on the transformed traces.